client/src/Components/postlist/postlist.js

```diff
@@ -14,25 +18,47 @@ class PostList extends React.Component{
             
             return res.json()
         })
-        .then((res)=>{
+        .then((res) => {
             console.log('got this',res)
             this.setState({post:res})
-        }) 
+        })
+    }
+    toggleDisplay = () => {
+        this.setState({
+            display:!this.state.display
+        })
+    }
+    addNewPost = async (recPost) => {
+        let newPost = this.state.post
+        let newObj = [recPost]
+        newPost = newObj.concat(newPost)
+        this.setState({
+            post:newPost
+        })
+        this.toggleDisplay()
+
     }
     render(){
         let ele = null;
         console.log(this.state.post)
-        if(this.state.post)
+        if(this.state.post && !this.state.display)
         {
             ele = this.state.post.map((item) => {
                 console.log(item._id,'item');
-                return <Post id={item._id}date={item.createdAt} owner={item.owner} location={item.location} caption = {item.caption} likes={item.likes} img={item.content} />
+                return <Post key={item.at} id={item._id} date={item.date} owner={item.owner} location={item.location} caption = {item.caption} likes={item.likes} img={item.content} />
             })
 
         }
-        return <ul>
+        else{
+            ele = <Upload toggleDisplay={this.toggleDisplay} addNewPost={this.addNewPost}/>
```

Comment:
Was the link not working here?

Reply:
Sir actually it had some state management and react routing didnt allow me to send functions from parent to child and then call it from child to parent so i did this

Reply:
Interesting.

Reply:
I tried the link way but it is not working it is getting stuck

Reply:
Ok let it be then

Reply:
https://ui.dev/react-router-v4-pass-props-to-link/

Did you try this way?

Reply:
sir actually when i am putting the link to that button the default behaviour of the post button is not working